File: 2025_08/11_Vowel_Balance/test_vowel_balance.py
import main_vowel_balance

"""
Given a string, determine whether the number of vowels in the first half of the string is equal to the number of vowels in the second half.
Vowels are: a, A, e, E, i, I, o, O, u, U
If there's an odd number of characters in the string, ignore the center character.
examples:
    1. is_balanced("racecar") should return True.
    2. is_balanced("Lorem Ipsum") should return True.
    3. is_balanced("Kitty Ipsum") should return False.
    4. is_balanced("string") should return False.
    5. is_balanced(" ") should return True.
    6. is_balanced("abcdefghijklmnopqrstuvwxyz") should return False.
    7. is_balanced("123A#b!E&*456-o.U") should return True.
"""

if __name__ == "__main__":
    # Test 1:
    if main_vowel_balance.is_balanced("racecar"):
        print("Test 1: PASSED")
    else:
        print("Test 1: FAILED")

    # Test 2:
    if main_vowel_balance.is_balanced("Lorem Ipsum"):
        print("Test 2: PASSED")
    else:
        print("Test 2: FAILED")

    # Test 3:
    if not main_vowel_balance.is_balanced("Kitty Ipsum"):
        print("Test 3: PASSED")
    else:
        print("Test 3: FAILED")

    # Test 4:
    if not main_vowel_balance.is_balanced("string"):
        print("Test 4: PASSED")
    else:
        print("Test 4: FAILED")

    # Test 5:
    if main_vowel_balance.is_balanced(" "):
        print("Test 5: PASSED")
    else:
        print("Test 5: FAILED")

    # Test 6:
    if not main_vowel_balance.is_balanced("abcdefghijklmnopqrstuvwxyz"):
        print("Test 6: PASSED")
    else:
        print("Test 6: FAILED")

    # Test 7:
    if main_vowel_balance.is_balanced("123A#b!E&*456-o.U"):
        print("Test 7: PASSED")
    else:
        print("Test 7: FAILED")

File: 2025_08/11_Vowel_Balance/test_vowel_balance.js
import  { isBalanced }  from "./main_vowel_balance.js";

/*
Given a string, determine whether the number of vowels in the first half of the string is equal to the number of vowels in the second half.
Vowels are: a, A, e, E, i, I, o, O, u, U
If there's an odd number of characters in the string, ignore the center character.
examples:
    1. is_balanced("racecar") should return True.
    2. is_balanced("Lorem Ipsum") should return True.
    3. is_balanced("Kitty Ipsum") should return False.
    4. is_balanced("string") should return False.
    5. is_balanced(" ") should return True.
    6. is_balanced("abcdefghijklmnopqrstuvwxyz") should return False.
    7. is_balanced("123A#b!E&*456-o.U") should return True.
*/

// --------------------------------------------------------------------------------------
// Test cases

let test_1 = String(isBalanced("racecar"))
let test_2 = String(isBalanced("Lorem Ipsum"))
let test_3 = String(isBalanced("Kitty Ipsum"))
let test_4 = String(isBalanced("string")) 
let test_5 = String(isBalanced(" "))
let test_6 = String(isBalanced("abcdefghijklmnopqrstuvwxyz"))
let test_7 = String(isBalanced("123A#b!E&*456-o.U"))

console.log(`Test 1: \n Ergebnis: ${test_1} \n Erwartet: true \n -------------------`)
console.log(`Test 2: \n Ergebnis: ${test_2} \n Erwartet: true \n -------------------`)
console.log(`Test 3: \n Ergebnis: ${test_3} \n Erwartet: false \n -------------------`)
console.log(`Test 4: \n Ergebnis: ${test_4} \n Erwartet: false \n -------------------`)
console.log(`Test 5: \n Ergebnis: ${test_5} \n Erwartet: true \n -------------------`)
console.log(`Test 6: \n Ergebnis: ${test_6} \n Erwartet: false \n -------------------`)
console.log(`Test 7: \n Ergebnis: ${test_7} \n Erwartet: true \n -------------------`)

// --------------------------------------------------------------------------------------
